frontend/app/src/store/blockchain/index.ts

```diff
@@ -113,46 +113,87 @@ export const useBlockchainStore = defineStore('blockchain', () => {
       return;
     }
 
-    try {
-      const addr = await Promise.all(
-        payload.map(data => addAccount(blockchain, data))
-      );
-      const filteredAddresses = addr.filter(add => add.length > 0);
-      if (blockchain === Blockchain.ETH && modules) {
-        await enableModule({
-          enable: modules,
-          addresses: filteredAddresses
-        });
+    const registeredAddresses: string[] = [];
+    const promiseResult = await Promise.allSettled(
+      filteredPayload.map(data => addAccount(blockchain, data))
+    );
+
+    const failedPayload: AccountPayload[] = [];
+
+    if (filteredPayload.length === 1) {
+      if (promiseResult[0].status === 'rejected') {
+        throw promiseResult[0].reason;
       }
-      resetDefi();
-      resetDefiStatus();
-      if (blockchain === Blockchain.ETH) {
-        await fetchDetected(filteredAddresses);
+    }
+
+    promiseResult.forEach((res, index) => {
```

Comment:
I will merge as it is but do you think it is possible to add some tests fo this flow after that?